pkg/operator/ceph/object/json_helpers.go

```diff
@@ -60,6 +70,10 @@ func setObjProperty[T string | []string | map[string]interface{} | []interface{}
 		if last {
 			// last path segment: set result and return prev value
 			prevVal, ok := obj[p]
+			if shouldExist && !ok {
+				// not exists:
+				return prev, nil
+			}
```

Comment:
I would expect a bool called `shouldExist` to cause the function to return an error if the field doesn't exist. Is there a better naming for this? Maybe `setIfExists`?

Reply:
@arttor this is also bringing up some related questions for me, and I'd like to clarify my understanding.

From reading the overall code, it seems like there is no error returned if the field Rook tries to set doesn't exist. Maybe I am misreading the code; if so let me know. If I'm reading correctly, is this for a specific reason?

IMO, it's probably safest for users for Rook to return an error if it tries to set a field using `setObjProperty()` if the field doesn't exist. Presumably, that case would happen if (1) RGW removes a field in the future, which we want to make sure we handle properly in code if needed, or (2) a dev makes a typo/mistake in a future Rook update.

Being able to catch (2) would be good, though not critical. Being able to catch (1) could be good to help make sure Rook's handling of RGW zone configs is always aware of RGW changes. I don't expect (1) to happen often, but catching it could be important.

Reply:
done. now it retruns error if pool not exists and the error is logged with INFO level.